boards/apollo3/lora_things_plus/Cargo.toml

```diff
@@ -19,3 +19,12 @@ apollo3 = { path = "../../../chips/apollo3" }
 capsules-core = { path = "../../../capsules/core" }
 capsules-extra = { path = "../../../capsules/extra" }
 capsules-system = { path = "../../../capsules/system" }
+
+[features]
+default = []
+
+# This feature enables support for the ATECC508A Cryptographic Co-Processor
+# Breakout. If you connect one of these
+# (https://www.sparkfun.com/products/15573) via the I2C Qwiic connector you
+# should enable this feature.
+atecc508a = []
```

Comment:
I think this is a great use of our newly established configuration boards concept. You can convert the `lora_things_plus` board into a bin+lib crate, and then create a configuration board which extends this base board with your peripheral.

I'm personally not opposed to `cfg`s in boards as much as I am for all other crates, but even here the fact that we don't test all (or actually, any) permutations of features means this is subject to bit rot.

And seeing that you have global `use` statements depend on this `cfg` makes me really skittish. Those are really hard to keep in sync & free of warnings without CI that tests all permutations of features.

Reply:
I was trying to avoid a configuration board for this. I think there are good use cases for the config boards, but I don't feel this is one of them.

In this case we are just enabling an off board peripheral. As it's only additive I feel it's a good use for a `cfg`. It also will hopefully allow adding things in the future. On top of that a `cfg` board is extra maintenance and the 3 apollo3 boards are already hard to keep in sync.

I agree about not being tested though. What if I ensure they are tested in the CI?

Reply:
This is tested in the CI now